File: Reserva-Hotel-com-Front-main/ReservaHotelFront/src/app/Hotel.ts
import { Avaliacao } from './Avaliacao';
import { Dono } from './Dono';
import { Cidade } from './Cidade';

export class Hotel {
    idHotel: number = 0;
    nome: string = '';
    endereco: string = '';
    avaliacoes: Avaliacao | undefined;
    descricao: string = '';
    listaDeQuartos: string = '';
    donoId: number = 0;
    dono: Dono | undefined;
    cidade: Cidade | undefined;
    numeroTotalDeQuartos: number = 0;
    numeroDeQuartosDisponiveis: number = 0;
}

File: Reserva-Hotel-com-Front-main/ProjetoCs-Reserva_de_Hotel-main/ReservaDeHotel/Models/Hotel.cs
using System.ComponentModel.DataAnnotations;

namespace ReservaDeHotel.Models;
public class Hotel
{
    [Key]
    public int IDHotel { get; set; }
    public string? Nome { get; set; }
    public string? Endereco { get; set; }
    public List<Avaliacao>? ListaAvaliacoes { get; set;}
    public int? ComodidadeId { get; set; }
    public Comodidade? Comodidade {get;set;}
    public string? Descricao { get; set; }
    public string? ListaDeQuartos { get; set; }
    public int NumeroTotalDeQuartos { get; set; }
    public int NumeroDeQuartosDisponiveis { get; set; }
    public List<Cidade>? Cidades { get; set;}
    public List<Dono>? Donos { get; set;}
}
